src/components/certificates/CertificateForm.tsx

```diff
@@ -1,17 +1,23 @@
-import { useCallback, useState } from 'react';
+import { useCallback, useContext, useEffect, useState } from 'react';
 import { useNavigate } from 'react-router-dom';
 
 import { ReactComponent as CloseSVG } from '../../assets/images/close.svg';
 import { ReactComponent as SearchSVG } from '../../assets/images/search.svg';
-import { addNewCertificate } from '../../data/db';
+import { CertificatesContext } from '../../contexts/certificatesContext';
+import { addNewCertificate, updateCertificate } from '../../data/db';
 import { Certificate, CertificateType } from '../../types/types';
+import formatValue from '../../utils/formatInputValue';
 import InputField from '../Form/InputField';
 import SelectField from '../Form/SelectFIeld';
 import SVGIcon from '../SVGIcon/SVGIcon';
 
 import './certificateForm.css';
 
-const NewCertificateForm: React.FC = () => {
+interface CertificateFormProps {
+  certificate?: Certificate;
+}
+
+const CertificateForm: React.FC<CertificateFormProps> = ({ certificate }) => {
   const navigate = useNavigate();
 
   const initialFormState: Certificate = {
```

Comment:
This is constant, not depending on any component data so it should be out of component